lib/Finance/Quote/Union.pm

```diff
@@ -109,41 +113,29 @@ sub unionfunds
         $info{$fund, "success"}  = 0;
         $info{$fund, "errormsg"} = "No data returned";
       }
-    }
-  }
-  else
-  {
-    foreach my $fund (@funds)
-    {
-      $info{$fund, "success"}  = 0;
-      $info{$fund, "errormsg"} = "HTTP error";
-    }
-  }
+	}
 
   return wantarray() ? %info : \%info;
-}
-
-# UNION provides a csv file named preise.csv containing the prices of all
-# their funds for the most recent business day.
 
-sub unionurl
-{
-  return "https://www.union-investment.de/preise.csv";
 }
+# UNION provides a csv file named historische-preise.csv on
+# <https://www.union-investment.de/fonds_depot/fonds-finden/preise-berechnen#HistorischeTagespreise>
+# containing the prices of a selction of all their funds for a selected period.
+
 
-1;
+__END__
 
 =head1 NAME
 
-Finance::Quote::Union	- Obtain quotes from UNION (Zurich Financial Services Group).
+Finance::Quote::Union	- Obtain quotes from UNION (Union Investment).
 
 =head1 SYNOPSIS
 
     use Finance::Quote;
 
     $q = Finance::Quote->new;
 
-    %stockinfo = $q->fetch("unionfunds","975788");
+    %stockinfo = $q->fetch("unionfunds","DE0008491002");
```

Comment:
Perhaps mention somewhere the change from german WKNs to ISIN?

Reply:
> Perhaps mention somewhere the change from german WKNs to ISIN?

You are right, I added that and the https: